src/pk/ecc/ecc_ssh_ecdsa_encode_name.c

```diff
@@ -0,0 +1,57 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+#include "tomcrypt_private.h"
+
+/**
+   @file ecc_ssh_ecdsa_encode_name.c
+   Curve/OID to SSH+ECDSA name string mapping per RFC5656
+   Russ Williams
+*/
+
+int ecc_ssh_ecdsa_encode_name(char *buffer, unsigned long buflen, const ecc_key *key)
+{
+   int err;
+
+   /* Check for three named curves: nistp256, nistp384, nistp521 */
+   if (ecc_cmp_oid_str("1.2.840.10045.3.1.7", key) == CRYPT_OK) {
+      /* nistp256 - secp256r1 - OID 1.2.840.10045.3.1.7 */
+
+      if (buflen <= 19) { err = CRYPT_BUFFER_OVERFLOW; goto error; }
```

Comment:
Maybe rather `buflen <= 20` - according to `man 3 strcpy`: The `strcpy()` function copies the string pointed to by src, **including the terminating null byte**.

Reply:
Yeah, the string is 19 chars long, so we need 19+1 to store it... if the buffer is 19 or fewer, it's too small. I figured "<= 19" was slightly clearer than "< 20" or "< 19 + 1", but none are particularly great.

What about:
```
   const char *name = "ecdsa-sha2-nistp256";
   if (buflen <= strlen(name)) { err = ... }
   XSTRCPY(buffer, name);
```